src/others/components/map/RegionsSource.tsx

```diff
@@ -0,0 +1,111 @@
+                              searchParams.get('regions_split_by_date') :
+                              process.env.REACT_APP_REGIONS_SPLIT_BY_DATE) === '1';
+
+  const activeDateFilter = filterContext.getActiveFilterItems("Dates")[0];
+  const layerFilterDate = (regionsSplitByDate ?
+                           ["boolean", "true"] :
+                           ["==", ["get", "date"], ["string", activeDateFilter]]);
+
+  const activeCategoryFilters = filterContext.getActiveFilterItems("Categories");
+
+  useEffect(() => {
+    if (!showRegions || !mapRef || !mapRef.current) return;
+    const allRegionsWithMeta = [];
+    for (const date of dates) {
+      if (regionsSplitByDate && date !== activeDateFilter) continue;
+      const regionToCount: {[id: string]: number} = {};
+      var maxVal = 0;
+      if (aidRequests) {
+        for (const request of aidRequests) {
+          if (request.date !== date) continue;
+          // TODO: support multi category
+          if (activeCategoryFilters.length > 0 && request.category_id !== activeCategoryFilters[0]) continue;
+          const city = translateLocation(request.city_id);
+          if (!city) continue;
+          const region_id = city.region_id;
+          if (!(region_id in regionToCount)) {
+            regionToCount[region_id] = 0;
+          }
+          regionToCount[region_id] = regionToCount[region_id] + request.requested_amount;
+          maxVal = Math.max(maxVal, regionToCount[region_id]);
+        }
+      }
+      for (const region of adminRegions) {
+        const res = regionsSplitByDate ? region : Object.assign({}, region);
+        if (res.properties) {
+          res.properties = Object.assign({}, res.properties);
+          res.properties.date = date;
+          if (res.properties.shapeID in regionToCount) {
+            res.properties.normalized_amount = regionToCount[res.properties.shapeID] / maxVal;
+          } else {
+            res.properties.normalized_amount = 0;
+          }
+        }
+        allRegionsWithMeta.push(res);
+      }
+    }
+    const regionsGeo: FeatureCollection<Geometry, GeoJsonProperties> = {
+      type: "FeatureCollection",
+      features: allRegionsWithMeta,
+    };
+    if (mapRef.current.getSource('state')) {
+      (mapRef.current.getSource('state') as GeoJSONSource).setData(regionsGeo);
+    }
+  }, [mapRef,
+      JSON.stringify(activeCategoryFilters),
+      regionsSplitByDate ? activeDateFilter : '',
+      JSON.stringify(dates),
+      JSON.stringify(aidRequests),  // This looks crazy, is there a better way?
+      translateLocation,
+      mapLoaded]);
+
+
+  if (showRegions) {
+   return <Source id="state" type="geojson" key="states" data={{type: "FeatureCollection", features: []}}>
+              <Layer id="state-borders" type="line" layout={{}} paint={{"line-color": "black", 'line-width': 1}} />
+              <Layer id="state-fills" type="fill" filter={layerFilterDate} layout={{}} 
+              paint={{
```

Comment:
What's the guideline for splitting styles into a separate file?

Because few line:107, there's also a style definition, should it also go to a separate file?